apps/website/auto-api.ts

```diff
@@ -63,93 +64,99 @@ type ParsedProps = {
-      if (lol.name === 'subComponentName' && subComponentName != lol.node.text) {
-        subComponentName = lol.node.text;
+    match.captures.forEach((capture) => {
+      // Statements are ordered as they appear in capture array
+      if (capture.name === 'subComponentName' && subComponentName !== capture.node.text) {
+        subComponentName = capture.node.text;
       }
-      if (lol.name === 'comment') {
-        //this removes the comment syntax
-        const justText = lol.node.text.replaceAll(/[/*]/g, '');
+      if (capture.name === 'comment') {
+        // This removes the comment syntax
+        const justText = capture.node.text.replace(/[/*]/g, '').trim();
         parsedProps.comment = justText;
       }
-
-      if (lol.name === 'prop') {
-        parsedProps.prop = lol.node.text;
+      if (capture.name === 'prop') {
+        parsedProps.prop = capture.node.text;
       }
-
-      if (lol.name === 'type') {
-        parsedProps.type = lol.node.text;
-        if (subComponentName === topKey(last)) {
-          last[topKey(last)].push(parsedProps);
+      if (capture.name === 'type') {
+        parsedProps.type = capture.node.text;
+        const lastKey = last ? Object.keys(last)[0] : '';
+        if (subComponentName === lastKey) {
+          last![lastKey].push(parsedProps);
         } else {
           parsed.push({ [subComponentName]: [parsedProps] });
         }
       }
     });
   });
 
-  const completeSubComponent: SubComponent = { [component_name[1]]: parsed };
+  const completeSubComponent: SubComponent = { [componentName]: parsed };
   ref.push(completeSubComponent);
   return ref;
 }
 
 function writeToDocs(fullPath: string, componentName: string, api: ComponentParts) {
   if (fullPath.includes('kit-headless')) {
-    const relDocPath = `../website/src/routes//docs/headless/${componentName}`;
+    const relDocPath = `../website/src/routes/docs/headless/${componentName}`;
     const fullDocPath = resolve(__dirname, relDocPath);
-    const dirPath = fullDocPath.concat('/auto-api');
+    const dirPath = resolve(fullDocPath, 'auto-api');
 
     if (!fs.existsSync(dirPath)) {
-      fs.mkdirSync(dirPath);
+      fs.mkdirSync(dirPath, { recursive: true });
     }
     const json = JSON.stringify(api, null, 2);
-    const hacky = `export const api=${json}`;
+    const exportedApi = `export const api = ${json};`;
 
     try {
-      fs.writeFileSync(dirPath.concat('/api.ts'), hacky);
-      console.log('auto-api: succesfully genereated new json!!! :)');
+      fs.writeFileSync(resolve(dirPath, 'api.ts'), exportedApi);
+      console.log('auto-api: successfully generated new JSON!');
     } catch (err) {
-      return;
+      console.error('Error writing API file:', err);
     }
   }
 }
+
 function loopOnAllChildFiles(filePath: string) {
-  const childComponentRegex = /\/([\w-]*).tsx$/.exec(filePath);
-  if (childComponentRegex === null) {
+  const childComponentMatch = /[\\/](\w[\w-]*)\.tsx$/.exec(filePath);
```

Comment:
good catch on escaping the dot. I'm still curious why the AI insist on having windows support  though